### DISCUSSION TITLE
Help Setting up trigger/workflow

### DISCUSSION THREAD

**Discussion by hasantahir** [zenangst/KeyboardCowboy#506] [Q&A]
Hi there,

I am not sure whether this will be called a workflow or a trigger, but let me explain what I intend to accomplish.

Say in an app called Latest, I want to check if there are any updates available for the apps that are installed on the machine:

<img width="813" alt="Screenshot 2024-03-04 at 17 19 08" src="https://github.com/zenangst/KeyboardCowboy/assets/11236148/42c1bfd0-4612-42f5-96eb-1d9981cfb1a9">

In Keyboard Cowboy, I have the following sequence through which I wish to access the `update all` button shown above:

<img width="1232" alt="Screenshot 2024-03-04 at 17 19 12" src="https://github.com/zenangst/KeyboardCowboy/assets/11236148/53c46d51-d90d-42a2-ba46-ff7a158f6d3b">

When I run the workflow by clicking the `Run Concurrent` button, nothing happens. Can you please tell me what am I doing wrong? Thanks

**Comment by zenangst** [comment]
@hasantahir I think what you want here is to change the actual trigger to run when `Latest` becomes active. But even if that is correct, there seems that you have found a small pesky little bug that needs squashing because I can get it working locally either.

Until the bug is fixed, you could rely on a menubar command to achieve the same result. It can look something like this:

<img width="1040" alt="image" src="https://github.com/zenangst/KeyboardCowboy/assets/57446/1890a701-d7db-4580-b3c7-593bdd6def7a">

**Comment by zenangst** [comment]
@hasantahir did some further investigation and I figure out what was going on here.
The bug will be fixed in the next version of Keyboard Cowboy :)

Thanks for bringing this to my attention ❤️

**Reply by hasantahir** [reply]
Thanks for getting back and not a problem. I have tried your suggestion but am still not able to make it work on my end. Waiting for the update 😎.

**Reply by zenangst** [reply]
@hasantahir did you update the trigger to run when the app is front most? I saw that your config was setup differently

**Comment by zenangst** [comment]
@hasantahir would you mind taking 3.23.1 for a spin to verify if this is fixed now?
https://github.com/zenangst/KeyboardCowboy/releases/tag/3.23.1

**Reply by hasantahir** [reply]
Hi @zenangst 

Thanks for getting back. I am experiencing the same behaviour as before. I have tried your configuration as well:
<img width="1012" alt="Screenshot 2024-03-15 at 22 51 27" src="https://github.com/zenangst/KeyboardCowboy/assets/11236148/a28a0314-e55f-4206-afb5-b84cd95c3d7f">

Mine:
<img width="1012" alt="Screenshot 2024-03-15 at 22 49 45" src="https://github.com/zenangst/KeyboardCowboy/assets/11236148/85f27093-2a77-4f05-9513-3a4596430372">

**Reply by zenangst** [reply]
Odd, works just like it should for me 🤔 

I would only check the `When in front most` on the trigger. I don't think you want to trigger the workflow when the application is closed. And I think `Launched` and `When in front most` might collide.

`When in front most` will be triggered at launch as well, so I think that is the only option that you want.

<img width="1012" alt="Screenshot 2024-03-16 at 11 31 11" src="https://github.com/zenangst/KeyboardCowboy/assets/57446/ba0c6b69-3e63-4722-81dc-99a5917aa897">

**Reply by zenangst** [reply]
@hasantahir found some additional fixes for application triggers that got released in version 3.23.2

**Reply by hasantahir** [reply]
Hi @zenangst 

Thanks for chasing this. I have still no luck in making this work. Is it possible to share your workflow? Like through a JSON file?

**Reply by zenangst** [reply]
There is no easy way to share workflows right now, it is on my never-ending todo list 😞
